### DISCUSSION TITLE
Alternatives for `get_subordinate_users` and `get_subordinate_groups`

### DISCUSSION THREAD

**Discussion by jrief** [django-cms/django-cms#7854] [Polls]
In our setup, users are never created by other users. Instead they are created by an external authentication system. The foreign key `created_by` in model `PageUser` hence always points onto a dummy user without any privileges. 

Therefore the two functions `cms.utils.permissions.get_subordinate_users` and `cms.utils.permissions.get_subordinate_groups` can't deliver any useful results, because all users are subordinate of that dummy user. I would therefore like to make those two functions configurable. I already created a prototype and named those two configuration directives `CMS_USER_SUBORDINATES ` and `CMS_GROUP_SUBORDINATES `.

Is there any objection to this feature?

We also might have to think about making the model `PageUser` optional. Without the possibility of creating and fetching subordinates, that model doesn't always make sense.

Making these two functions configurable can be achieved by applying this patch:

https://github.com/jrief/django-cms/commit/80cc54dd5bdda7accbbf82a333042bd472ec66bb

**Comment by marksweb** [comment]
I'm always in favour of making things more flexible so if we can maintain the current functionality while allowing you to customise this, then that's great.
